src/geode/geometry/nn_search.cpp

```diff
@@ -86,6 +122,151 @@ namespace geode
             return results;
         }
 
+        typename NNSearch< dimension >::ColocatedInfo colocated_index_mapping(
```

Comment:
If you are able to template this function and put it in an anonymous namesapce (or even in this class), it will be simpler